src/wpsmon.c

```diff
@@ -59,13 +59,26 @@ enum seen_flags {
 	SEEN_FLAG_COMPLETE = 2,
 };
 static unsigned seen_count;
+
+static void list_clean() {
+	unsigned i;
```

Comment:
this can be done in a single line with memset(seen_list, 0, sizeof seenlist);

Reply:
actually i think we can just reset the seen_count to 0. elements >= seen_count are never compared.

Reply:
seen_list[i].probes and seen_list[i].flags need reset too.
```C
memset(seen_list, 0, sizeof seen_list);
seen_count = 0;
```